state/indexer/block/kv/kv.go

```diff
@@ -317,6 +320,13 @@ LOOP:
 			if err != nil {
 				continue LOOP
 			}
+
+			if matchEvents && qr.Key != types.BlockHeightKey {
+				keyHeight, err := parseHeightFromEventKey(it.Key())
+				if err != nil || !checkHeightConditions(heightInfo, keyHeight) {
```

Comment:
We need to pass the range here because if we have another range (say `foo > 300`) it should return events only if both `foo > 300` and the height of the event is within the height range . This was previously ignored when match.events = 1.